full_cluster_replacement_test.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from longevity_test import LongevityTest
+from sdcm import prometheus
+from sdcm.sct_events.system import InfoEvent
+from sdcm.utils import loader_utils
+
+LOGGER = logging.getLogger(__name__)
+
+
+class FullClusterReplacement(LongevityTest, loader_utils.LoaderUtilsMixin):
+    """
+    The purpose of these tests is replacement of all nodes in the ScyllaDB.
+    Cluster Replacement QA task: https://github.com/scylladb/qa-tasks/issues/1164
+    Cluster replacement test plan: https://docs.google.com/document/d/1aKCTs6SiH9ORDYvkAQ8upOpVwZqaxsW9VfbJqKG63g0/edit
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.metrics_srv = prometheus.nemesis_metrics_obj()
+        self.stress_queue = []
+
+    def replace_node(self, node_to_be_removed, dc_idx=1):
+        # add node
+        InfoEvent(message='Replace Node - Starting to add node to cluster').publish()
+        new_node = self.db_cluster.add_nodes(1, dc_idx=dc_idx, enable_auto_bootstrap=True)[0]  # add node
+        self.db_cluster.wait_for_init(node_list=[new_node], timeout=900,
+                                      check_node_health=False)
+        InfoEvent(message='Replace Node - Ended adding node to cluster').publish()
+        self.db_cluster.wait_for_nodes_up_and_normal(nodes=[new_node])
+        self.monitors.reconfigure_scylla_monitoring()
+        # Decommission node to be replaced
+        InfoEvent(message='Replace Node - Decommission node: {}'.format(node_to_be_removed)).publish()
+        self.db_cluster.decommission(node_to_be_removed)
+
+    def verify_stress_threads_after_node_replacement(self):
+        for stress in self.stress_queue:
+            self.verify_stress_thread(cs_thread_pool=stress)
+
+    def test_multidc_rolling_full_cluster_replacement_add_then_remove_nodes(self):
+        InfoEvent("Starting cluster replacement test...")
+
+        keyspace_num = self.params.get('keyspace_num')
+        stress_cmd = self.params.get('stress_cmd')
+
+        # Start the load
+        self.run_pre_create_schema()
+        self.run_prepare_write_cmd()
+        # self.run_post_prepare_cql_cmds() Uncomment when MV solved: Ref: https://github.com/scylladb/scylladb/issues/15717
+        self.assemble_and_run_all_stress_cmd(self.stress_queue, stress_cmd, keyspace_num)
+
+        # Run node replacement procedure for all nodes and verify data
+        number_of_nodes_in_each_dc = int(len(self.db_cluster.nodes)/2)
+        for i, node in enumerate(self.db_cluster.nodes):
+            if i >= number_of_nodes_in_each_dc:
+                dc_idx = 0
+            else:
+                dc_idx = 1
+            self.replace_node(node_to_be_removed=node, dc_idx=dc_idx)
+            self.verify_stress_threads_after_node_replacement()
+
+        # Full Repair and major compaction
```

Comment:
where this need comes from? Why there's no cleanup?

Reply:
Added cleanup to the `replace_node` method.
This came from Fabio's request in the test plan.

Reply:
Worth to mention in a comment that this is not a must in terms of the replacement procedure.
It's another part of the test to test a full repair and major compaction on a replaced cluster.

Reply:
Added a comment